### PR TITLE
ci-process-bootstrap, add cleanup for any outstanding watchers or workers

### PR DESCRIPTION
**Pull request by lsh-0**

### CODE REVIEW

File: bin/ci-import [hunk 1]
```diff
@@ -17,5 +17,13 @@ echo "Starting import"
...
-. bin/ci-expected-results
+./bin/ci-expected-results || {
+    echo "-----"
+
+    echo "failed expected results. pausing and trying again"
+    
+    sleep 5
+    
+    ./bin/ci-expected-results
+}
```
Comment (lsh-0) [inline bin/ci-import:28]: revert

File: bin/ci-reindex [hunk 2]
```diff
@@ -18,5 +18,13 @@ echo "Finished launching reindexing"
...
-. bin/ci-expected-results
+./bin/ci-expected-results || {
+    echo "-----"
+
+    echo "failed expected results. pausing and trying again"
+    
+    sleep 5
+    
+    ./bin/ci-expected-results
+}
```
Comment (lsh-0) [inline bin/ci-reindex:29]: revert

File: project_tests.sh [hunk 3]
```diff
@@ -1,6 +1,8 @@
 #!/usr/bin/env bash
 set -e
 
+sudo systemctl stop elasticsearch
```
Comment (lsh-0) [inline project_tests.sh:4]: revert

File: src/Search/Api/Elasticsearch/MappedElasticsearchClient.php [hunk 4]
```diff
@@ -67,6 +67,10 @@ public function searchDocuments(array $query) : QueryResponse
 
     public function getDocumentById($id, $index = null)
     {
+        #$fh = fopen("/tmp/mylog", "a");
+        #fwrite($fh, "got index " . $index . " and given index " . $this->index . " for id " . $id . "\n");
+        #fclose($fh);
+
```
Comment (lsh-0) [inline src/Search/Api/Elasticsearch/MappedElasticsearchClient.php:73]: revert

**Comment by NuclearRedeye** [comment]
I built upon the foundations here as part of [PR](https://github.com/elifesciences/search/pull/366) which has now been merged, hence closing this PR.

**Pull request closed without merging**